src/main/java/sg/ncl/MainController.java

```diff
@@ -334,6 +339,68 @@ public String maintainance() {
         return "maintainance";
     }
 
+    @RequestMapping(value = "/networkTool", method = RequestMethod.GET)
+    public String networkTopologyTool() {
+        return "network_diagram";
+    }
+
+    @RequestMapping(value = "/networkTool", method = RequestMethod.POST)
+    public @ResponseBody String networkTopologyAnalysis(@RequestParam("jsonText") String jsonText) {
+        String nsfilename = null;
+        JSONObject jsonObject = new JSONObject();
+        StringBuilder logBuilder = new StringBuilder();
+        String filename = networkToolProperties.getTemp() + System.currentTimeMillis() + ".json";
+        try (FileWriter fw = new FileWriter(filename)) {
+            fw.write(jsonText);
+        } catch (IOException ioe) {
+            log.error(ioe.toString());
+        }
+        try {
+            ProcessBuilder pb = new ProcessBuilder(
+                    networkToolProperties.getCommand(),
+                    networkToolProperties.getVersion(),
+                    networkToolProperties.getProgram(),
+                    networkToolProperties.getOption(),
+                    filename);
+            Process p = pb.start();
+            BufferedReader br = new BufferedReader(new InputStreamReader(p.getInputStream()));
+            for (String line = br.readLine(); line != null; line = br.readLine()) {
+                if (line.contains("Produce NSfile")) {
+                    Pattern pattern = Pattern.compile("\\[([^]]+)\\]");
+                    Matcher matcher = pattern.matcher(line);
+                    if (matcher.find()) {
+                        nsfilename = matcher.group(1) + "/NSfile.txt";
+                    }
+                }
+                logBuilder.append(line).append("&#010;");
+            }
+        } catch (IOException ioe) {
+            log.error(ioe.toString());
+        } finally {
+            Path filePath = Paths.get(filename);
+            if (Files.exists(filePath)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().exists()" method [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3725)